gammapy/maps/geom.py

```diff
@@ -355,19 +355,18 @@ def __init__(self, nodes, interp="lin", name="", node_type="edges", unit=""):
         self._nbin = nbin
 
     def __eq__(self, other):
-        if isinstance(other, self.__class__):
-            return (
-                np.allclose(self._nodes, other._nodes)
+        """Test axis equality. Absolute and relative tolerances of 1e-6 are used"""
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+
+        # TODO: implement an allclose method for MapAxis and call it here
+        other_quantity = other.edges * other.unit
```

Comment:
Naively I would have expected to just check unit and values separately. I.e. have "equal" semantics on all object data members, not "equivalency" semantics.

I guess it's a matter of taste.

I don't see that there's really a use case where you'd need to check `axis1 == axis2` where both are "equivalent" quantities, but not "equal". So I'm +1 to check unit and values separately, that's useful e.g. in serialisation tests.

Reply:
OK done